### PR TITLE
refactor: extend table context usage

### PR DESCRIPTION
**Pull request by cbt1**
A prototype where we use React Context to avoid prop-drilling some base properties (nebula props) that are shared by multiple components.

This version has only made changes and virtual scroll mode and is for now just a PoC.

I looked a bit at performance and how many times each component renders in comparision with the solution on `main` and so far there is no obvious difference. The gain seem to be a cleaner "tree" for component properties.

### CODE REVIEW

File: src/table/types.ts [hunk 1]
```diff
@@ -78,6 +78,15 @@ export interface ContextValue {
   selectionDispatch: SelectionDispatch;
   hoverIndex: number;
   setHoverIndex: React.Dispatch<React.SetStateAction<number>>;
+  baseProps: {
+    selectionsAPI: ExtendedSelectionAPI;
+    layout: TableLayout;
+    model?: EngineAPI.IGenericObject;
```
Comment (haxxmaxx) [inline src/table/types.ts:84]: I think we should make sure model is defined in the index file

**Review by haxxmaxx — COMMENTED** [review]
Looks good! Are there any additional renders? Could you try it for the pagination as well? 😇

File: src/table/types.ts (hunk 1 above)
Comment (cbt1) [reply src/table/types.ts:84]: I had to make it optional as the paginiation table does not pass the model

**Comment by cbt1** [comment]
> Looks good! Are there any additional renders? Could you try it for the pagination as well? 😇

Not that I could see from my testing. Performance wise feelt the same but some actual number on that would be nice

File: src/table/types.ts (hunk 1 above)
Comment (haxxmaxx) [reply src/table/types.ts:84]: ah I see, then we shouldn't change anything in the index 🙃

**Comment by haxxmaxx** [comment]
> > Looks good! Are there any additional renders? Could you try it for the pagination as well? 😇
> 
> Not that I could see from my testing. Performance wise feelt the same but some actual number on that would be nice

Is it possible to use the profiler for that?

**Comment by cbt1** [comment]
> Looks good! Are there any additional renders? Could you try it for the pagination as well? 😇

I can, but not in this PR. As that would be to much of a change I think. There are going be a few tests that needs to be wrapped in a provider now, which is a bit of a downside.

File: src/table/components/virtualized-table/__tests__/Table.spec.tsx [hunk 2]
```diff
@@ -14,32 +15,24 @@ describe('<Table />', () => {
   let pageInfo: PageInfo;
   let paginationNeeded: boolean;
   let model: EngineAPI.IGenericObject;
-  let theme: ExtendedTheme;
+  // let theme: ExtendedTheme;
```
Comment (a-m-dev) [inline src/table/components/virtualized-table/__tests__/Table.spec.tsx:18]: ops!

File: jest.config.js [hunk 3]
```diff
@@ -4,4 +4,5 @@ module.exports = {
   setupFilesAfterEnv: ['./jest.setup.js'],
   modulePathIgnorePatterns: ['<rootDir>/mashup-example'],
   transformIgnorePatterns: ['<rootDir>/node_modules/(?!(@qlik-trial)/)'],
+  coveragePathIgnorePatterns: ['src/__test__/generate-test-data.ts', 'src/__test__/test-with-providers.tsx']
```
Comment (cbt1) [inline jest.config.js:7]: We should move the test utils files to some other location but I'll do that in another PR as I dont want all those changed test files in this PR as well

**Review by a-m-dev — APPROVED** [review]
LGTM 🚀

**Comment by haxxmaxx** [comment]
is the description still accurate, that this is considered a PoC/prototype?

**Comment by cbt1** [comment]
> is the description still accurate, that this is considered a PoC/prototype?

No, I've updated it now if someone would ever want to go back and look at this PR.

**Pull request merged**